schedule.py

```diff
@@ -214,18 +268,22 @@ def create_model(professors, sections, semesters):
     # soft constraints
 
     # assign classes according to prof preference
+    # maximize the number of sections offered
+    # can assign importance with coefficient
     model.Maximize(sum(
         classes[(prof_name, section_name)] * professor.prefers(section.course)
         for prof_name, professor in professors.items()
         for section_name, section in sections.items()
+    ) + sum(
+        classes[(prof_name, section_name)]
+        for prof_name, professor in professors.items()
+        for section_name, section in sections.items()
```

Comment:
Without this, the solver will only schedule courses that are preferred by profs. If should be fine if professors' preferences cover enough classes.

Reply:
:+1:

Reply:
Whoops, replied to the wrong thing. There needs to be a separate column in the courses tab that say whether a course *must* be offered versus whether it *could* be. This can be added once the time constraint stuff is done.